File: 696-count-binary-substrings/count-binary-substrings.java
class Solution {
    public int countBinarySubstrings(String s) {
        int zeros=0, ones=0;
        int last=-1,count=0;
        for(char c:s.toCharArray()) {
            if(c == '0') {
                if(last == 1) zeros=0;
                zeros++;
                last=0;
            } else {
                if(last == 0) ones=0;
                ones++;
                last=1;
            }
            if((last==1 && zeros >= ones) || (last==0 && ones >= zeros)) count++;
        }
        return count;
    }
}

File: 696-count-binary-substrings/count-binary-substrings.cpp
class Solution {
public:
    int countBinarySubstrings(string s) {
        int z = 0, o = 0;
        int last = -1, count = 0;
        for(char c : s) {
            if(c == '0') {
                if(last == 1) z = 0;
                z++;
                last=0;
            } else {
                if(last == 0) o=0;
                o++;
                last=1;
            }
            if((last==1 && z >= o) || (last==0 && o >= z)) count++;
        }
        return count;
    }
};